controllers/topics-controller.js

```diff
@@ -0,0 +1,50 @@
+const { selectTopics, readTopics, readTopicsBySlug } = require('../models/topics-model');
+
+const getAllTopics = (request, response, next) => {
+    selectTopics().then((topics) => {
+        response.status(200).send(topics);  
+    })
+    .catch((err) => {
+        next(err);
+    });
+};
+
+const getTopicBySlug = (request, response, next) => {
+    const {slug} = request.params;
+    if(!isNaN(slug)){
+        throw {status: 400};
+    };
+
+readTopicsBySlug(slug)
+    .then((topics) => {
+        if(topics.length === 0){
+        throw {status: 404};
+        }
+        response.status(200).send(topics);   
+    })
+    .catch((err) => {
+        next(err);
+    });
+};
+
+const getTopicByQuery = (request, response, next) => {
+    const {order_by} = request.params
```

Comment:
As mentioned previously, no queries are expected at this stage or on this endpoint